nixos/modules/services/misc/pufferpanel.nix

```
@@ -0,0 +1,401 @@
+          list of packages as a reference:
+          ```Nix
+          services.pufferpanel.extraPackages =
+            with pkgs; [
+              coreutils
+              bash
+              gawk
+              gzip
+              gnutar
+              unzip
+              xz
+              wget
+              curl
+              git
+              mono
+              jre
+            ]
+            ++
+            map lib.getLib [
+              stdenv.cc.cc.lib # for libstdc++
+              zlib
+              icu
+              openssl
+              libgdiplus
+              curl
+              sqlite
+            ];
+          ```
+
+          Note, however, that this option should only be used as a last resort.
+          It is otherwise recommended to configure required packages via
+          {option}`extraPackages` with this option disabled.
+
+          [templates]: https://github.com/PufferPanel/templates
+        '';
+      };
+
+      dynamicLinker = lib.mkOption {
+        type = lib.types.nullOr lib.types.package;
+        default = stdenvDynamicLinker;
+        defaultText = lib.literalMD ''
+          Package containing {file}`lib` directory with symlink to dynamic linker
+          specified in `''${pkgs.stdenv.cc}/nix-support/dynamic-linker`.
+        '';
+        description = lib.mdDoc ''
+          Dynamic linker made available at `/lib` (and `/lib64` on 64-bit
+          systems) for running binaries downloaded by PufferPanel. These
+          directories are private to the service’s filesystem namespace.
+        '';
+      };
+    };
+
+    environment = lib.mkOption {
+      type = lib.types.attrsOf lib.types.str;
+      default = { };
+      example = lib.literalExpression ''
+        {
+          PUFFER_DAEMON_CONSOLE_BUFFER = "1000";
+          PUFFER_DAEMON_CONSOLE_FORWARD = "true";
+          PUFFER_TOKEN_PUBLIC = "https://panel.example.com/auth/publickey";
+        }
+      '';
+      description = lib.mdDoc ''
+        Environment variables to set for the service. Secrets should be
+        specified using {option}`environmentFile`.
+
+        Refer to the [PufferPanel source code][] for the list of available
+        configuration options. Variable name is an upper-cased configuration
+        entry name with underscores instead of dots, prefixed with `PUFFER_`.
+        For example, `panel.settings.companyName` entry can be set using
+        {env}`PUFFER_PANEL_SETTINGS_COMPANYNAME`.
+
+        [PufferPanel source code]: https://github.com/PufferPanel/PufferPanel/blob/master/config/entries.go
+      '';
+    };
+
+    environmentFiles = lib.mkOption {
+      type = lib.types.listOf lib.types.path;
+      default = [ ];
+      description = lib.mdDoc ''
+        Files to load environment variables from. Loaded variables override
+        values set in {option}`environment`.
+      '';
+    };
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
```

```suggestion
      default = false;
```
we don't open firewalls by default, also people want probably to run this behind nginx or similar